src/core/src/bootstrap/ConfigurationFactory.py

```diff
@@ -155,18 +165,21 @@ def new_bootstrap_configuration(config_env, log_file_path, real_record_path, rec
 
     def new_prod_configuration(self, package_manager_name, package_manager_component):
         """ Base configuration for Prod V2. """
+
         configuration = {
             'config_env': Constants.PROD,
             'package_manager_name': package_manager_name,
             'lifecycle_manager': {
-                'component': LifecycleManager,
+                'component': self.lifecycle_manager_component,
                 'component_args': ['env_layer', 'execution_config', 'composite_logger', 'telemetry_writer'],
                 'component_kwargs': {}
             },
             'status_handler': {
                 'component': StatusHandler,
                 'component_args': ['env_layer', 'execution_config', 'composite_logger', 'telemetry_writer'],
-                'component_kwargs': {}
+                'component_kwargs': {
+                    'vm_context': self.vm_context
```

Comment:
vm_cloud_type is what came to mind first, but open to discussion.

(where cloud type refers to Arc or Azure obviously)